doc/user/basic.rst

```diff
@@ -435,6 +443,134 @@ Puppet, etc.), upgrade considerations differ somewhat:
    the same checks as when rolling out upgrades.
 
 
+.. _protocol-key-encryption:
+
+Protocol key encryption
+-----------------------
+
+Various routing protocols include configurable passwords, authentication
+strings, or other sorts of key strings that will be referred to in this
+section as "protocol keys." Historically, FRR has written these protocol keys
+as plain text to configuration files and to the craft interface.
+
+In some environments, it is preferable to encrypt these strings in saved
+and displayed configuration, because these data are sometimes shared widely
+as part of organizational debugging or coordination processes. The protocol
+key encryption feature is designed to prevent inadvertent disclosure of
+protocol keys in configuration written to files and to the craft interface.
+
```

Comment:
From discussion:
This feature protects against disclosure of keys when the show configuration output or frr.conf file is shared.  It does not protect against disclosure when the <XYZ file> is shared along with the encrypted information.

Reply:
add:
It also permits backing up and restoring the configuration file.